pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go

```diff
@@ -554,14 +554,29 @@ func handleRangefeedError(ctx context.Context, err error) (rangefeedErrorInfo, e
 	}
 }
 
+// catchup alloc is a catchup scan allocation.
+type catchupAlloc func()
+
+// Release implements limit.Reservation
+func (a catchupAlloc) Release() {
+	a()
+}
+
 func acquireCatchupScanQuota(
 	ctx context.Context, ds *DistSender, catchupSem *limit.ConcurrentRequestLimiter,
 ) (limit.Reservation, error) {
```

Comment:
Consider setting the return type to `catchupAlloc` instead of `limit.Reservation`. Also, since it's passed down the stack, replace the type there too.
Technically it's the same interface, so it's compatible, but hiding it behind `limit.Reservation` can be confusing.

Reply:
Sure.